sdk/typescript/test/unit/cryptography/multisig.publickey.test.ts

```diff
@@ -0,0 +1,411 @@
+				{ publicKey: pk3, weight: 3 },
+			],
+		});
+
+		const multiSigPublicKey2 = MultiSigPublicKey.fromPublicKeys({
+			threshold: 4,
+			publicKeys: [
+				{ publicKey: pk1, weight: 1 },
+				{ publicKey: pk2, weight: 2 },
+				{ publicKey: pk3, weight: 3 },
+			],
+		});
+
+		const multiSigPublicKey3 = MultiSigPublicKey.fromPublicKeys({
+			threshold: 3,
+			publicKeys: [
+				{ publicKey: pk1, weight: 1 },
+				{ publicKey: pk2, weight: 2 },
+			],
+		});
+
+		const multiSigPublicKey4 = MultiSigPublicKey.fromPublicKeys({
+			threshold: 3,
+			publicKeys: [
+				{ publicKey: pk1, weight: 1 },
+				{ publicKey: pk2, weight: 2 },
+				{ publicKey: pk3, weight: 3 },
+			],
+		});
+
+		expect(multiSigPublicKey1.equals(multiSigPublicKey2)).toEqual(false);
+		expect(multiSigPublicKey1.equals(multiSigPublicKey3)).toEqual(false);
+		expect(multiSigPublicKey1.equals(multiSigPublicKey4)).toEqual(true);
+	});
+
+	it('`toRawBytes()` should return correct array representation', async () => {
+		const multiSigPublicKey = MultiSigPublicKey.fromPublicKeys({
+			threshold: 3,
+			publicKeys: [
+				{ publicKey: pk1, weight: 1 },
+				{ publicKey: pk2, weight: 2 },
+				{ publicKey: pk3, weight: 3 },
+			],
+		});
+
+		const data = new Uint8Array([0, 0, 0, 5, 72, 101, 108, 108, 111]);
+
+		const sig1 = await k1.signPersonalMessage(data);
+		const sig2 = await k2.signPersonalMessage(data);
+
+		const multisig = multiSigPublicKey.combinePartialSignatures([sig1.signature, sig2.signature]);
+		const rawBytes = fromB64(multisig).slice(134);
+
+		expect(multiSigPublicKey.toRawBytes()).toEqual(rawBytes);
+	});
+
+	it('`getPublicKeys()` should return correct publickeys', async () => {
+		const multiSigPublicKey = MultiSigPublicKey.fromPublicKeys({
+			threshold: 3,
+			publicKeys: [
+				{ publicKey: pk1, weight: 1 },
+				{ publicKey: pk2, weight: 2 },
+				{ publicKey: pk3, weight: 3 },
+			],
+		});
+
+		expect(multiSigPublicKey.getPublicKeys()).toEqual([
+			{ publicKey: pk1, weight: 1 },
+			{ publicKey: pk2, weight: 2 },
+			{ publicKey: pk3, weight: 3 },
+		]);
+	});
+
+	it('`toSuiAddress()` should return correct sui address associated with multisig publickey', async () => {
+		const multiSigPublicKey = MultiSigPublicKey.fromPublicKeys({
+			threshold: 3,
+			publicKeys: [
+				{ publicKey: pk1, weight: 1 },
+				{ publicKey: pk2, weight: 2 },
+				{ publicKey: pk3, weight: 3 },
+			],
+		});
+
+		const maxLength = 1 + (64 + 1) * MAX_SIGNER_IN_MULTISIG + 2;
+		const tmp = new Uint8Array(maxLength);
+		tmp.set([0x03]);
+		tmp.set(builder.ser('u16', 3).toBytes(), 1);
+		let i = 3;
+		for (const { publicKey, weight } of multiSigPublicKey.getPublicKeys()) {
+			const bytes = publicKey.toSuiBytes();
+			tmp.set(bytes, i);
+			i += bytes.length;
+			tmp.set([weight], i++);
+		}
+		const multisigSuiAddress = normalizeSuiAddress(
+			bytesToHex(blake2b(tmp.slice(0, i), { dkLen: 32 })),
+		);
+
+		expect(multiSigPublicKey.toSuiAddress()).toEqual(multisigSuiAddress);
```

Comment:
generally, I prefer hard-coding an expected address over re-implementing the tested functionality in the test.  Not a  huge issue, but if for example if something changes about what bytesToHex or blake2b returns  (unlikely), the test would silently continue to pass with the changed values, since the implementation and are using the very similar implementations.

I've noticed this across a number of tests here where the test and implementation both just implement the same logic in slightly different ways.  I wouldn't consider this a blocker, just wanted to mention it

Reply:
On it